internal/kafka/internal/services/cluster_placement_strategy.go

```diff
@@ -18,6 +18,8 @@ func NewClusterPlacementStrategy(clusterService ClusterService, dataplaneCluster
 	var clusterSelection ClusterPlacementStrategy
 	if dataplaneClusterConfig.IsDataPlaneManualScalingEnabled() {
 		clusterSelection = &FirstSchedulableWithinLimit{dataplaneClusterConfig, clusterService, kafkaConfig}
+	} else if dataplaneClusterConfig.IsDataPlaneAutoScalingEnabled() {
+		clusterSelection = &FirstReadyWithCapacity{clusterService, kafkaConfig}
 	} else {
 		clusterSelection = &FirstReadyCluster{clusterService}
 	}
```

Comment:
Instead of `if ... else if ... else` I think a switch is more readable

```suggestion
	switch {
		case dataplaneClusterConfig.IsDataPlaneManualScalingEnabled():
			clusterSelection = &FirstSchedulableWithinLimit{dataplaneClusterConfig, clusterService, kafkaConfig}
		case dataplaneClusterConfig.IsDataPlaneAutoScalingEnabled():
			clusterSelection = &FirstReadyWithCapacity{clusterService, kafkaConfig}
		default:
			clusterSelection = &FirstReadyCluster{clusterService}
	}
```